vpype_gcode/gwrite.py

```diff
@@ -38,247 +42,58 @@ def invert_axis(
-                'move': '\t\t{\n\t\t\t"X": %d,\n\t\t\t"Y": %d\n\t\t}',
-                'line': ',\n\t\t{\n\t\t\t"X": %d,\n\t\t\t"Y": %d\n\t\t}',
-                'postblock': '\n\t\t],\n',
-                'postlayer': '\t},\n',
-            }  # Json is nearly validated has extra ',' on the last post-block and post-layer.
-
-    }
-    if version in writers:
-        writer = writers[version]
-        if 'header' in writer and header is None:
-            header = writer['header']
-        if 'move' in writer and move is None:
-            move = writer['move']
-        if 'line' in writer and line is None:
-            line = writer['line']
-        if 'preblock' in writer and preblock is None:
-            preblock = writer['preblock']
-        if 'postblock' in writer and postblock is None:
-            postblock = writer['postblock']
-        if 'prelayer' in writer and prelayer is None:
-            prelayer = writer['prelayer']
-        if 'postlayer' in writer and postlayer is None:
-            postlayer = writer['postlayer']
-        if 'footer' in writer and footer is None:
-            footer = writer['footer']
-        if 'unit' in writer and unit is None:
-            unit = writer['unit']
-        if 'negate_x' in writer and negate_x is None:
-            negeate_x = writer['negate_x']
-        if 'negate_y' in writer and negate_y is None:
-            negate_y = writer['negate_y']
-        if 'invert_x' in writer and invert_x is None:
-            invert_x = writer['invert_x']
-        if 'invert_y' in writer and invert_y is None:
-            invert_y = writer['invert_y']
-        if 'relative' in writer and relative is None:
-            relative = writer['relative']
-    if header is not None:
-        header = header.replace('\\n', '\n')
-    if move is not None:
-        move = move.replace('\\n', '\n')
-    if line is not None:
-        line = line.replace('\\n', '\n')
-    if preblock is not None:
-        preblock = preblock.replace('\\n', '\n')
-    if postblock is not None:
-        postblock = postblock.replace('\\n', '\n')
-    if prelayer is not None:
-        prelayer = prelayer.replace('\\n', '\n')
-    if postlayer is not None:
-        postlayer = postlayer.replace('\\n', '\n')
-    if footer is not None:
-        footer = footer.replace('\\n', '\n')
-    if relative is None:
-        relative = False
-    if negate_x is None:
-        negate_x = False
-    if negate_y is None:
-        negate_y = False
-    if invert_x is None:
-        invert_x = False
-    if invert_y is None:
-        invert_y = False
+def gwrite(document: vp.Document, filename: str, profile: str):
+    gwrite_config = vp.CONFIG_MANAGER.config["gwrite"]
+
+    # If no profile was provided, try to use a default
+    if not profile:
+        # Try to get the default profile from the config
+        if "default_profile" in gwrite_config:
+            profile = gwrite_config["default_profile"]
+        else:
+            raise Exception(
```

Comment:
You might want to check click's doc and use the exception type that turns into an user error message instead of the usual, arcane-to-the-layman backtrack.

Reply:
Done, fixed this and replaced it the a click.BadParameter exception